crates/ty_python_semantic/src/types.rs

```diff
@@ -1216,6 +1212,31 @@ impl<'db> Type<'db> {
                 | Type::ModuleLiteral(_),
             ) => false,
 
+            (Type::NominalInstance(_) | Type::ProtocolInstance(_), Type::Callable(_)) => {
+                let call_symbol = self
+                    .member_lookup_with_policy(
```

Comment:
This looks wrong -- shouldn't we be looking up `__call__` only on the class, not on the instance?

It looks like this pre-exists this PR, and we probably shouldn't add more changes to this PR, but maybe we should add a TODO here, while we're here?

Reply:
this is okay because of the fact that we're passing the `NO_INSTANCE_FALLBACK` policy:

https://github.com/astral-sh/ruff/blob/1274521f9fda19507921c85442d0ca991b67b846/crates/ty_python_semantic/src/types.rs#L192-L200

Reply:
Ah right, I forgot the details of how this works